legion/engine/physics/physx/systems/physx_physics_system.cpp

```diff
@@ -5,38 +5,32 @@ namespace legion::physics
 {
     using namespace physx;
 
-    struct physXStatics
+    struct PhysxStatics
     {
-        static size_type sSelfInstanceCounter;
+        inline static size_type selfInstanceCounter = 0;
 
-        static PxFoundation* foundation;
-        static PxPvd* pvd;
-        static PxDefaultCpuDispatcher* dispatcher;
-        static PxPhysics* phyxSDK;
+        inline static PxFoundation* foundation = nullptr;
+        inline static PxPvd* pvd = nullptr;
+        inline static PxDefaultCpuDispatcher* dispatcher = nullptr;
+        inline static PxPhysics* phyxSDK = nullptr;
 
-        static PxDefaultAllocator defaultAllocator;
-        static PxDefaultErrorCallback defaultErrorCallback;
+        inline static PxDefaultAllocator defaultAllocator;
+        inline static PxDefaultErrorCallback defaultErrorCallback;
     };
 
-    size_type physXStatics::sSelfInstanceCounter = 0;
+    using PS = PhysxStatics;
 
-    PxFoundation* physXStatics::foundation = nullptr;
-    PxPvd* physXStatics::pvd = nullptr;
-    PxDefaultCpuDispatcher* physXStatics::dispatcher = nullptr;
-    PxPhysics* physXStatics::phyxSDK = nullptr;
-
-    PxDefaultAllocator physXStatics::defaultAllocator;
-    PxDefaultErrorCallback physXStatics::defaultErrorCallback;
-    
     //PhysX PVD Debugger Related
     constexpr const char* pvdHost = "127.0.0.1";
     constexpr size_type defaultPVDListeningPort = 5425;
     constexpr size_type defaultPVDHostTimeout = 10;
 
     void PhysXPhysicsSystem::setup()
     {
-        physXStatics::sSelfInstanceCounter++;
+        const std::lock_guard<std::mutex> setupLock(m_setupShutdownMutex);
 
+        PS::selfInstanceCounter++;
+        
         lazyInitPhysXVariables();
         setupDefaultScene();
 
```

Comment:
also i don't think createProcess should be called within the critical section of the lock